libminifi/include/core/yaml/YamlConfiguration.h

```diff
@@ -245,11 +245,11 @@ class YamlConfiguration : public FlowConfiguration {
    *
    * @param yamlNode a pointer to the YAML::Node that will be checked for the
    *                   presence of an idField
-   * @param idField  the string of the name of the idField to check for. This
+   * @param field  the string of the name of the idField to check for. This
    *                   is optional and defaults to 'id'
    * @return         the parsed or generated UUID string
    */
-  std::string getOrGenerateId(YAML::Node *yamlNode, const std::string &idField = "id");
+  std::string getOrGenerateField(YAML::Node *yamlNode, const std::string &field = "id");
```

Comment:
What is the end goal of this PR? It seems that you want to generalize the function name to eventually be used elsewhere.

Reply:
End goal is outlined in https://issues.apache.org/jira/browse/MINIFICPP-359. getOrGenerateField is used in this PR for name now, not just id: `std::string name = getOrGenerateField(&connectionNode, "name");`

Reply:
I see that you added the assignment, but I don't see how we are generating fields other than ID fields with that function. In this case it is okay because name will exist at this point in the code by virtue of the checkRequiredFields call. 

In the case where someone else uses it, there is little to indicate other than const std::string &field = "id" that they will get a uuid field value per the conditional in getOrGenerateField. In my opinion they are getting or generating an ID in the current function, hence my confusion.

Reply:
The checkRequiredField call was removed:

```diff
-        checkRequiredField(&connectionNode, "name",
 -        CONFIG_YAML_CONNECTIONS_KEY);
```

So the current behavior is that name can be left unspecified. In that case it will be generated.